### ISSUE TITLE
snowflake_resource_monitor SQL compilation error

### ISSUE THREAD

**Issue by jrobison-sb** [Snowflake-Labs/terraform-provider-snowflake#1652]
**Provider Version**

```
+ provider registry.terraform.io/snowflake-labs/snowflake v0.56.0
```

**Terraform Version**

```
Terraform v1.4.0
on darwin_arm64
```

**Describe the bug**

`SQL compilation error` upon modifying a `snowflake_resource_monitor` resource.

**Expected behavior**

The `snowflake_resource_monitor` resource should be modified without error.

**Code samples and commands**

I have a `snowflake_resource_monitor` which I've been using successfully for many months. I now have a need to modify that resource. Here are my plan diffs:

```
  # module.qa_application.snowflake_resource_monitor.monitor[0] will be updated in-place
  ~ resource "snowflake_resource_monitor" "monitor" {
        id                         = "QA-SFMONTOR"
        name                       = "QA-SFMONTOR"
      ~ notify_triggers            = [
          - 100,
        ]
      ~ suspend_immediate_triggers = [
          + 100,
        ]
      ~ suspend_triggers           = [
          - 100,
          + 90,
        ]
        # (4 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

And if I apply that, I'll get this error:

```
001003 (42000): SQL compilation error:
syntax error line 1 at position 50 unexpected 'ON'.
syntax error line 1 at position 64 unexpected 'DO'.
syntax error line 1 at position 75 unexpected 'ON'.
syntax error line 1 at position 90 unexpected 'DO'.
syntax error line 1 at position 110 unexpected '<EOF>'.
```

Here is the underlying query as seen in the debug logs:

```
2023-03-24T10:41:35.606-0400 [INFO]  provider.terraform-provider-snowflake_v0.56.0: 2023/03/24 10:41:35 [DEBUG] sql-conn-exec [query ALTER RESOURCE MONITOR "QA-SFMONTOR" SET TRIGGERS ON 90 PERCENT DO SUSPEND ON 100 PERCENT DO SUSPEND_IMMEDIATE err 001003 (42000): SQL compilation error: syntax error line 1 at position 50 unexpected 'ON'. syntax error line 1 at position 64 unexpected 'DO'. syntax error line 1 at position 75 unexpected 'ON'. syntax error line 1 at position 90 unexpected 'DO'. syntax error line 1 at position 110 unexpected '<EOF>'. duration 105.219ms args {}]: timestamp=2023-03-24T10:41:35.606-0400
```

**Comment by bennylu2** [comment]
I think this was fixed in recent patch #1643 but has not been released yet

**Comment by jrobison-sb** [comment]
This bug was resolved for me by upgrading the provider version (to 0.67.0, though an earlier version of >= 0.60.0 might also have worked).
